nntrainer/layers/layer_node.cpp

```diff
@@ -129,6 +129,55 @@ class SharedFrom : public Name {
  *
  */
 LayerNode::~LayerNode() = default;
+  
+/**
+ * @brief get the compute engine property from property string vector
+ *  : default is CPU
+ * @return LayerComputeEngine Enum : CPU, GPU, QNN
+ *
+ */
+ml::train::LayerComputeEngine
+getComputeEngine(const std::vector<std::string> &props) {
+  for (auto &prop : props) {
+    std::string key, value;
+    int status = nntrainer::getKeyValue(prop, key, value);
+    if (nntrainer::istrequal(key, "engine")) {
+      constexpr const auto data =
+        std::data(props::ComputeEngineTypeInfo::EnumList);
+      for (uint i = 0; i < props::ComputeEngineTypeInfo::EnumList.size(); ++i) {
+        if (nntrainer::istrequal(value.c_str(),
+                                 props::ComputeEngineTypeInfo::EnumStr[i])) {
+          return data[i];
+        }
+      }
+    }
+  }
+
+  return ml::train::LayerComputeEngine::CPU;
+}
+
+/**
+ * @brief get the compute engine property from property string vector
+ *  : default is CPU
+ * @return LayerComputeEngine Enum : CPU, GPU, QNN
+ *
+ */
+ml::train::LayerComputeEngine
+getComputeEngine(const std::vector<std::string> &props) {
+  for (auto &prop : props) {
```

Comment:
Is this common way to have to to enumerate all properties and filter on its value?

Such code hints that possible future improvement would be:
- to executed `nntrainer::getKeyValue`  once for all properties creating mapping
Example: std::map/std::unordered_map<property_key, property_value>/our implementation of future std::flat_map
- switching all such functions to accept such mapping rather doing O(n) filtering.

In the code execution it costs O(N*M) right now where N is number of specified properties and M is number of possible properties.

Other possilble benefits:
- It is harder to read and write such filtering probably;
- Would give an opportunity to 'case-normalize' keys of such collection